src/config/watcher.rs

```diff
@@ -139,9 +141,8 @@ mod tests {
         tokio::time::timeout(timeout, fut).await.is_ok()
     }
 
-    #[tokio::test]
+    #[test_util::test]
```

Comment:
I feel like `test_util::test` doesn't give a great idea of what this macro does. Without looking at it, I'd expect that I'd need to add this to _every_ test.

I feel like I lack the knowledge to suggest a better name though :smile: Maybe `test_util::async_test`? Or, following Tokio's naming, `util::async_test`? Or simply `async_test` if we move it to the main crate.

Reply:
What if we call the crate `vector_test`?

Reply:
We can use `test_util::test` for <i>every</i> test because this macros initialize trace logs. I can change macros in a way that `Runtime` will not be created for not `async` tests. This sounds like magic from another side.. :thinking: maybe two macroses: `sync_test` and `async_test`?

Reply:
I'll change current name to `async_test` and add `sync_test`. I placed macros under `test_util::` because everything test related in this scope. Is there better place? `crate::async_test`?

Reply:
> We can use test_util::test for every test because this macros initialize trace logs

If this is the case, would it be possible to just have that automatically done rather than annotate every test?

I like `sync_test` and `async_test`.